internal/flow/internal/controller/loader.go

```diff
@@ -350,16 +356,25 @@ func (l *Loader) populateDeclareNodes(g *dag.Graph, declareBlocks []*ast.BlockSt
 			})
 			continue
 		}
-		// TODO: if node already exists in the graph, update the block
-		// instead of copying it.
-		node := NewDeclareNode(declareBlock)
-		if g.GetByID(node.NodeID()) != nil {
-			diags.Add(diag.Diagnostic{
-				Severity: diag.SeverityLevelError,
-				Message:  fmt.Sprintf("cannot add declare node %q; node with same ID already exists", node.NodeID()),
-			})
+
+		if diag, defined := blockAlreadyDefined(blockMap, id, declareBlock); defined {
+			diags = append(diags, diag)
 			continue
 		}
+
+		if exist := l.graph.GetByID(id); exist != nil {
+			node = exist.(*DeclareNode)
+			node.UpdateBlock(declareBlock)
+		} else {
+			node = NewDeclareNode(declareBlock)
+			if g.GetByID(node.NodeID()) != nil {
+				diags.Add(diag.Diagnostic{
+					Severity: diag.SeverityLevelError,
+					Message:  fmt.Sprintf("cannot add declare node %q; node with same ID already exists", node.NodeID()),
+				})
+				continue
```

Comment:
Is this check required anymore? It seems like the `blockAlreadyDefined` call would handle this case now.